tests/ts_simple_query_buffers_SUITE.erl

```diff
@@ -167,48 +167,129 @@ init_per_testcase(query_orderby_max_data_size_error, Cfg) ->
     Cfg;
 
 init_per_testcase(query_orderby_ldb_io_error, Cfg) ->
-    Node = hd(proplists:get_value(cluster, Cfg)),
-    QBufDir = filename:join([rtdev:node_path(Node), "data/query_buffers"]),
-    modify_dir_access(take_away, QBufDir),
+    Cfg;
+
+init_per_testcase(query_orderby_inmem2ldb, Cfg) ->
     Cfg;
 
 init_per_testcase(_, Cfg) ->
     Cfg.
 
 end_per_testcase(query_orderby_max_quanta_error, Cfg) ->
     Node = hd(proplists:get_value(cluster, Cfg)),
-    ok = rpc:call(Node, application, set_env, [riak_kv, max_query_quanta, 1000]),
+    ok = rpc:call(Node, application, set_env, [riak_kv, timeseries_query_max_quanta_span, 1000]),
     ok;
 
 end_per_testcase(query_orderby_max_data_size_error, Cfg) ->
     Node = hd(proplists:get_value(cluster, Cfg)),
     ok = rpc:call(Node, riak_kv_qry_buffers, set_max_query_data_size, [5*1000*1000]),
     ok;
 
-end_per_testcase(query_orderby_ldb_io_error, Cfg) ->
-    Node = hd(proplists:get_value(cluster, Cfg)),
-    QBufDir = filename:join([rtdev:node_path(Node), "data/query_buffers"]),
-    modify_dir_access(give_back, QBufDir),
+end_per_testcase(query_orderby_ldb_io_error, _Cfg) ->
     ok;
+
+end_per_testcase(query_orderby_inmem2ldb, _Cfg) ->
+    ok;
+
 end_per_testcase(_, Cfg) ->
     Cfg.
 
 
 query_orderby_max_quanta_error(Cfg) ->
+    ct:print("Testing error reporting (max_query_quanta) ...", []),
     Query = ts_qbuf_util:full_query(?TABLE, [{order_by, [{"d", undefined, undefined}]}]),
     ok = ts_qbuf_util:ack_query_error(Cfg, Query, ?E_QUANTA_LIMIT).
 
 query_orderby_max_data_size_error(Cfg) ->
+    ct:print("Testing error reporting (max_query_data_size) ...", []),
     Query1 = ts_qbuf_util:full_query(?TABLE, [{order_by, [{"d", undefined, undefined}]}]),
     ok = ts_qbuf_util:ack_query_error(Cfg, Query1, ?E_SELECT_RESULT_TOO_BIG),
     Query2 = ts_qbuf_util:full_query(?TABLE, [{order_by, [{"d", undefined, undefined}]}, {limit, 99999}]),
     ok = ts_qbuf_util:ack_query_error(Cfg, Query2, ?E_SELECT_RESULT_TOO_BIG).
 
 query_orderby_ldb_io_error(Cfg) ->
+    ct:print("Testing error reporting (IO error) ...", []),
+    Node = hd(proplists:get_value(cluster, Cfg)),
+
+    C = rt:pbc(Node),
+    %% force immediate creation of ldb instance (otherwise taking away permission will not take effect)
+    load_intercept(Node, C, {riak_kv_qry_buffers, [{{can_afford_inmem, 1}, can_afford_inmem_no}]}),
+
+    QBufDir = get_qbuf_dir(Node),
+    modify_dir_access(take_away, QBufDir),
+
     Query = ts_qbuf_util:full_query(?TABLE, [{order_by, [{"d", undefined, undefined}]}]),
-    ok = ts_qbuf_util:ack_query_error(Cfg, Query, ?E_QBUF_CREATE_ERROR).
+    Res = ts_qbuf_util:ack_query_error(Cfg, Query, ?E_QBUF_INTERNAL_ERROR),
+
+    modify_dir_access(give_back, QBufDir),
+    rt_intercept:clean(Node, riak_kv_qry_buffers),
+
+    ok = Res,
+    ok.
 
 
+%%
+%% 4. checking transition from inmem to ldb during data collection
+%% --------------------------------------
+
+query_orderby_inmem2ldb(Cfg) ->
```

Comment:
The test case in https://github.com/basho/riak_test/pull/1257/files#diff-095dac84f424d6a5f5dd43a19ac4c37eR257 emulates the conditions to cause the qbuf manager to behave functionally differently: (1) immediately write qbufs to leveldb, (2) hold the data completely in memory and (3) hold for some time, then dump it. Cases (1) and (2) are assured by the intercepted `can_afford_inmem` always returning `false` or `true`, respectively.

The naive measurements I provide are for illustrative purposes; as expected, the pure-inmem trial completes faster, but it is perf team business to estimate by how much.